MDANSE/Src/MDANSE/IO/MinimalPDBReader.py

```diff
@@ -74,113 +67,123 @@ def __init__(self, filename: str):
         """
 
-        coordinates = []
-        element_slice = atom_line_slice("element_symbol")
-        name_slice = atom_line_slice("atom_name")
-        posx_slice = atom_line_slice("pos_x")
-        posy_slice = atom_line_slice("pos_y")
-        posz_slice = atom_line_slice("pos_z")
-        residue_slice = atom_line_slice("residue_name")
-        residue_number_slice = atom_line_slice("residue_number")
+        element_slice = ATOM_LINE_RECORDS["element_symbol"]
+        name_slice = ATOM_LINE_RECORDS["atom_name"]
+        pos_slice = (ATOM_LINE_RECORDS[x] for x in ("pos_x", "pos_y", "pos_z"))
+        residue_slice = ATOM_LINE_RECORDS["residue_name"]
+        residue_number_slice = ATOM_LINE_RECORDS["residue_number"]
 
+        coordinates = []
         element_list = []
         name_list = []
         label_dict = {}
         clusters = {}
 
         for atom_number, atom_line in enumerate(atom_lines):
-            chemical_element = atom_line[element_slice].strip()
+
+            chemical_element = atom_line[element_slice].strip().capitalize()
             atom_name = atom_line[name_slice]
             processed_atom_name = atom_name[:2].strip()
-            if len(processed_atom_name) == 2:
-                if processed_atom_name[0].isnumeric():
-                    processed_atom_name = processed_atom_name[1].upper()
-                else:
-                    processed_atom_name = (
-                        processed_atom_name[0].upper() + processed_atom_name[1].lower()
-                    )
-            if len(chemical_element) == 2:
-                chemical_element = (
-                    chemical_element[0].upper() + chemical_element[1].lower()
-                )
-            backup_element = atom_line.rstrip().split()[-1]
-            backup_element2 = atom_line.split()[-2].strip()
+
+            if len(processed_atom_name) == 2 and processed_atom_name[0].isnumeric():
+                processed_atom_name = processed_atom_name[1]
+            processed_atom_name = processed_atom_name.capitalize()
+
+            backup_elements = atom_line.split()[-2:]
+
             if atom_name[-2:].isnumeric():
-                backup_element3 = atom_name[0]
-            else:
-                backup_element3 = "fail"
-            if backup_element in ATOMS_DATABASE.atoms:
-                element_list.append(backup_element)
-            elif backup_element2 in ATOMS_DATABASE.atoms:
-                element_list.append(backup_element2)
-            elif backup_element3 in ATOMS_DATABASE.atoms:
-                element_list.append(backup_element3)
-            elif chemical_element in ATOMS_DATABASE.atoms:
-                element_list.append(chemical_element)
-            elif processed_atom_name in ATOMS_DATABASE.atoms:
-                element_list.append(processed_atom_name)
+                backup_elements.append(atom_name[0])
+
+            for trial in (chemical_element, processed_atom_name, *backup_elements):
```

Comment:
Somewhere around here (but possibly a few lines before) the logic of the code got changed, and if I convert a Gromacs trajectory of a protein using this branch, it ends up containing helium and mercury where it was supposed to be hydrogen.

Should we add a test for it? I will share some example files with you.

Reply:
I changed the order of `backup_elements` to go at the end (as one might think they ought) but I can stick them back at the front again. Try this:
```suggestion
            for trial in (*backup_elements, chemical_element, processed_atom_name):
```